components/kyma-environment-broker/internal/environmentscleanup/service.go

```diff
@@ -59,25 +61,28 @@ func (s *Service) PerformCleanup() error {
 		runtimeID, ok := shoot.Annotations[shootAnnotationRuntimeId]
 		if !ok {
 			err = errors.New(fmt.Sprintf("shoot %q has no runtime-id annotation", shoot.Name))
-			log.Error(err)
-			result = multierror.Append(result, err)
+			s.logger.Error(err)
 			continue
 		}
 		runtimeIDsToDelete = append(runtimeIDsToDelete, runtimeID)
 	}
-	log.Infof("Runtime IDs to process: %v", runtimeIDsToDelete)
+	s.logger.Infof("Runtime IDs to process: %v", runtimeIDsToDelete)
+
+	if len(runtimeIDsToDelete) == 0 {
+		return result.ErrorOrNil()
+	}
 
 	instancesToDelete, err := s.getInstancesForRuntimes(runtimeIDsToDelete)
 	if err != nil {
-		log.Error(errors.Wrap(err, "while getting instance IDs for Runtimes"))
+		s.logger.Error(errors.Wrap(err, "while getting instance IDs for Runtimes"))
 		result = multierror.Append(result, err)
```

Comment:
Underlying logic in `getInstancesForRuntimes` has retries when accessing the database. Entering this code block means that we could not reach the db and get instances, so we can also return error immediately right here (with possible errors appended before in the loop)